galette/lib/Galette/Entity/Adherent.php

```diff
@@ -2414,4 +2416,89 @@ protected function getEventsPrefix(): string
     {
         return 'member';
     }
+
+
+    /**
+     * Get QR codes associated to member
+     *
+     * @return QrCode[]
+     */
+    public function getQrCodes(): array
+    {
+        global $routeparser, $login;
+
+        $qrcodes = [];
+
+        if (!$login->isAdmin() && !$login->isStaff() && !$login->isGroupManager()) {
+            //only admin, staff and group managers can get QR codes
+            return $qrcodes;
+        }
+
+        $qrcodes['vcard'] = new QrCode(
+            data: $this->getVCard()->serialize(),
+            label: $this->sname,
+            url: $routeparser->urlFor('memberVCard', ['id' => $this->id]),
+            logo: GALETTE_ROOT . '/includes/qr-logos/address-card-o.svg'
+        );
+
+        if (!empty($this->getEmail())) {
+            $qrcodes['email'] = new QrCode(
+                data: 'mailto:' . $this->getEmail(),
+                label: $this->getEmail(),
+                url: 'mailto:' . $this->getEmail(),
+                logo: GALETTE_ROOT . '/includes/qr-logos/envelope-o.svg'
+            );
+        }
+
+        if (!empty($this->phone)) {
+            $qrcodes['phone'] = new QrCode(
+                data: 'tel:' . $this->phone,
+                label: $this->phone,
+                url: 'tel:' . $this->phone,
+                logo: GALETTE_ROOT . '/includes/qr-logos/fax.svg'
+            );
+        }
+
+        if (!empty($this->gsm)) {
+            $qrcodes['gsm'] = new QrCode(
+                data: 'tel:' . $this->gsm,
+                label: $this->gsm,
+                url: 'tel:' . $this->gsm,
+                logo: GALETTE_ROOT . '/includes/qr-logos/mobile-phone.svg'
```

Comment:
Why not use the emojis included in Fomantic?

```suggestion
                logo: GALETTE_ROOT . '/webroot/themes/default/ui/themes/galette/assets/emojis/1f4f1.svg'
```